http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -213,14 +213,28 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
       paramMap <- F.fromEither(processListParameters(params))
       creatorOnly <- F.fromEither(processCreatorOnlyParameter(userInfo.userEmail, params, ctx.traceId))
       disks <- DiskServiceDbQueries.listDisks(paramMap._1, paramMap._2, creatorOnly, cloudContext).transaction
+      partition = disks.partition(_.cloudContext.isInstanceOf[CloudContext.Gcp])
       _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done DB call")))
-      diskAndProjects = disks.map(d =>
-        (GoogleProject(d.cloudContext.asString), d.samResource)
-      ) // TODO: update this to support Azure
-      samVisibleDisksOpt <- NonEmptyList.fromList(diskAndProjects).traverse { ds =>
+
+      gcpDiskAndProjects = partition._1.map(d => (GoogleProject(d.cloudContext.asString), d.samResource))
+      gcpSamVisibleDisksOpt <- NonEmptyList.fromList(gcpDiskAndProjects).traverse { ds =>
         authProvider
           .filterResourceProjectVisible(ds, userInfo)
       }
+
+      azureDiskAndProjects = disks.map(d => (GoogleProject(d.cloudContext.asString), d.samResource))
```

Comment:
Is this intended to work across all disks, not just one of the partitioned subsets?

Reply:
So the quick fix I’m doing here is to partition by type (azure vs gcp) and filter by project for GCP, and filter the old way for azure, then add them back together. This is copying the old way these disks were filtered and why I added back the `filterUserVisibleWithProjectFallback`, which is the old filtering function

Reply:
My confusion is that this is doing `disks.map` and not `partition._2.map` - is that intentional? Would that not do this over all disks, and not just Azure disks?

Reply:
on line 219 the same thing is done for gcp disks (partition._1)

Reply:
but didn't combine them for this map because once they're converted to this form `(GoogleProject, SamResourceId)`, I can't determine the cloud platform

Reply:
I'm sorry, I think I'm missing something. What I mean is what I see is

```
allDisks = listDisks()
partition = allDisks.partition(d.isGcp)
gcpDisksMapped = partition._1.map(foo)
azureDisksMapped = allDisks.map(foo)
```

the `azureDisksMapped` here is a map of all disks, right? Is it okay that we're double-processing the GCP disks that got pulled out specially in the earlier step?

Reply:
ohhhhhhh you're right, totally my bad, this is supposed to be `partition._2`

Reply:
I was too hasty with this fix

Reply:
fix PR: https://github.com/DataBiosphere/leonardo/pull/3641